pkg/kubectl/resize.go

```diff
@@ -33,23 +34,45 @@ type ResizePrecondition struct {
 	ResourceVersion string
 }
 
+// A PreconditionError is returned when a replication controller fails to match
+// the resize preconditions passed to kubectl.
 type PreconditionError struct {
 	Precondition  string
 	ExpectedValue string
 	ActualValue   string
 }
 
-func (pe *PreconditionError) Error() string {
+func (pe PreconditionError) Error() string {
 	return fmt.Sprintf("Expected %s to be %s, was %s", pe.Precondition, pe.ExpectedValue, pe.ActualValue)
 }
 
+// Types of ControllerResizeErrors
+const (
```

Comment:
I'd considered using iota but wanted to avoid forcing people to look it up should they enter this code. I'll update, since this feels more idiomatic.